src/utils/general/ParseTools.cpp

```diff
@@ -162,25 +82,46 @@ bool isHeaderLine(const string &line) {
 		return true;
 	}
 	
-	string tmp = line;
-	transform(tmp.begin(), tmp.end(), tmp.begin(), ::tolower);
-	//allow chr chrom to start a header line
-	if (memcmp(tmp.c_str(), "chrom", 5) == 0 && isspace(tmp[5]) && ! isdigit(tmp[6])) {
-		return true;
-	}
-	//allow chr chrom to start a header line
-	if (memcmp(tmp.c_str(), "chr", 3) == 0 && isspace(tmp[3]) && ! isdigit(tmp[4])) {
-		return true;
-	}
-	//UCSC file headers can also start with the words "browser" or "track", followed by a whitespace character.
-	if (memcmp(tmp.c_str(), "browser", 7) == 0) {
-		return true;
-	}
-	if (memcmp(tmp.c_str(), "track", 5) == 0) {
-		return true;
-	}
-	if (memcmp(tmp.c_str(), "visibility", 10) == 0) {
-		return true;
+	if(line.length() > 4) {
+		uint32_t peek = *(uint32_t*)line.c_str() | 0x20202020u;
```

Comment:
This makes unportable assumptions about endianness and alignment.

Reply:
I am not worry about this. I think bedtools is already unportable, so in this situation, I think it's reasonable for trading portability at this point for performance. In fact, we are not expecting this runs on cpu other than X86. In the future, we should refactor this completely.

Reply:
Which code already in bedtools do you think is unportable?

Reply:
Thanks for the comment, @jmarshall - @38 made that block dependent on the detection of X86 CPU.